internal/controller/runtime/fsm/runtime_fsm_apply_clusterrolebindings.go

```diff
@@ -58,6 +58,10 @@ func sFnApplyClusterRoleBindings(ctx context.Context, m *fsm, s *systemState) (s
 		"Cluster admin configuration complete",
 	)
 
+	if !s.instance.IsProvisioningCompletedStatusSet() {
```

Comment:
@mvshao : wouldn't it be better to move this step into a final last FMS-state? 

If we add later on another step for processing a cluster, it could happen that we set the `provisionedCompleted` step to READY even before the whole process was finished.

Reply:
@tobiscr the `sFnApplyClusterRoleBindings` state is our final FMS state. Before applying `k.Status.ProvisioningCompleted = true` on CR, in line 55-58 RuntimeCR is marked as `Ready`.

In my opinion, having a separate FSM state just for setting the status `ProvisioningCompleted` is not needed. Until now we are able to successfully add, edit and ,delete FSM states without changing the logic.